### PR TITLE
Issue 1633 -- Update hitbtc with backwards compatible

### PR DESCRIPTION
**Pull request by EricAnderson1000**
Please provide feedback on this approach.

@timmolter Reviewer
@npomfret Reviewer

This is a backwards compatible change that leaves all existing calls in place. New services will as best as possible keep existing interfaces.. but some method signatures will indeed change with the differences in the APIs.
What do you think about this approach? 

Adding tests to new services in package 'v2'

I could also commit smaller PRs for review with this approach.

### CODE REVIEW

File: xchange-hitbtc/src/main/java/org/knowm/xchange/hitbtc/v2/HitbtcExchange.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  private final SynchronizedValueFactory<Long> nonceFactory = new CurrentTimeNonceFactory();
+
+  private HitbtcMetaData hitbtcMetaData;
+
+  @Override
+  protected void initServices() {
+
+    marketDataService = new HitbtcMarketDataService();
+    tradeService = new HitbtcTradeService(this);
```
Comment (EricAnderson1000) [inline xchange-hitbtc/src/main/java/org/knowm/xchange/hitbtc/v2/HitbtcExchange.java]: Borrowing old services here until new ones are implemented.

File: xchange-hitbtc/src/main/java/org/knowm/xchange/hitbtc/v2/HitbtcExchange.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  @Override
+  public SynchronizedValueFactory<Long> getNonceFactory() {
+
+    return nonceFactory;
+  }
+
+  @Override
+  public void remoteInit() throws IOException {
+
+    HitbtcRestClient.INSTANCE.init(this);
```
Comment (EricAnderson1000) [inline xchange-hitbtc/src/main/java/org/knowm/xchange/hitbtc/v2/HitbtcExchange.java]: Moving the actual rest proxy to a singleton. In the older version we are creating this object 3 times for each service.

**Comment by npomfret** [comment]
I'll should have time to take a look tomorrow.

**Comment by EricAnderson1000** [comment]
@npomfret @timmolter Do you have time to look over this PR today?

**Comment by timmolter** [comment]
Can you not use Guava? We've managed to keep it out of XChnage for many years now. What features do you need?

**Comment by EricAnderson1000** [comment]
>Can you not use Guava? We've managed to keep it out of XChnage for many years now. What features do you need?

It's scope is test, which means it shouldn't be part of the jar artifact right? I am using it for it's functional map transformation in a test. If Java 8 were used, it wouldn't be needed for sure. If you still prefer it not be used in tests, I can remove it. Just let me know.

**Comment by timmolter** [comment]
I'd be open for updating to Java 8. Let me open up a specific issue to get people's opinion. 

Is this ready to be merged now? @npomfret comments?

**Comment by EricAnderson1000** [comment]
test this please

**Comment by EricAnderson1000** [comment]
If it's OK with everyone, lets merge this, and I will keep adding features to the new Hitbtc v2 package.

**Comment by npomfret** [comment]
Ok with me.  But can we be very quick to fix any regressions?

**Comment by EricAnderson1000** [comment]
>Ok with me. But can we be very quick to fix any regressions?

Of course. But I would be surprised if there were any.  Since existing code is left in place and only deprecated.

**Pull request merged**